### ISSUE TITLE
APNS doesn't work unless if tokens are added via loop

### ISSUE THREAD

**Issue by jamrader** [Ph3nol/NotificationPusher#123]
For some reason the notification will arrive if I do the following:

```
$devices = new DeviceCollection(

        );
        $devices->add(new Device('7f7640a6aafef467f4b85129f55003d1c41ed5f16d8d1245c2925934b1154218'));
        $devices->add(new Device('c9d95136d92115b252263b40a0b8cd7ebe85ejf3c068f22a269ef70796b2381f'));

```

But not via a loop:

```
$devices = new DeviceCollection(

        );
foreach($iosTokens as $val) {
                $dev = new Device("$val->devicetoken");
                $devices->add($dev);
            }
```

Any reason why that might be?

**Comment by jamrader** [comment]
It looks like the problem is if a single token of those added fails to send the push, the entire push (to all the devices, or at least those after the incorrect token) fails too. How do I prevent a single incorrect token from ruining the entire message?

**Comment by crissi** [comment]
Also have this problem

**Comment by Fomenko-Slava** [comment]
Also have this problem. How can I resolve it?
I write something like this:

$devices = PushNotification::DeviceCollection($devices);
PushNotification::app('ios')
    ->to($devices)
    ->send($push_message);

But if a single token from devices fails all devices after this fails too.

**Comment by leouze** [comment]
I know this is old but this might help. #111

**Comment by jesseforrest** [comment]
I am also noticing the same issue where if one token is no longer valid, that none of the push notifications will work. This almost entirely defeats the purpose of using the DeviceCollection. Is it required to first determine which tokens are not valid using the following:
https://github.com/Ph3nol/NotificationPusher/blob/master/doc/apns-adapter.md#feedback-example

I think this is high priority or at least documentation updated to explain that this can happen.

**Comment by seyfer** [comment]
Seems like a real issue. 
The problem is, that author of the lob doesn't have time and I don't have iOS device at the moment to test.

@jesseforrest, feedback returns some result only after some tokens were sent. It's not possible to test them before actually trying to send them.

**Comment by seyfer** [comment]
Fixed in #155
